### ISSUE TITLE
🐛 [Bug Report] [WoTLK] [Crash] Server crashes on out of memory

### ISSUE THREAD

**Issue by adellaci** [cmangos/issues#3140]
### Bug Details

Started server on 2022-09-11 12:21:20     "SERVER STARTUP TIME: 1 minutes 51 seconds"
Played on that day for about two hours, one player, one character
Played again on 2022-19-15 for 45 minutes, one player, one character
Server crashed on 2022-09-16 18:45:10, 

[mangosd (copy).csv](https://github.com/cmangos/issues/files/9591725/mangosd.copy.csv)

Saved all other logs. Did not post them, as they are large, link below:

Server_2022-09-11_12-21-20.log         41.7 MB https://www.dropbox.com/s/w531cetjwbzab6s/Server_2022-09-11_12-21-20.log?dl=0
world_2022-09-11_12-21-20.log         183.2 MB https://www.dropbox.com/s/9zgh37o9hzgy82s/world_2022-09-11_12-21-20.log?dl=0
Char_2022-09-11_12-21-20.log              1.2 kB      https://www.dropbox.com/s/s1lnwgjyyg8r8uh/Char_2022-09-11_12-21-20.log?dl=0
2022-09-12_logSQL.sql                          531 byte   https://www.dropbox.com/s/98z5jwqopnvtf8x/2022-09-12_logSQL.sql?dl=0
DBErrors.log                                         578.8 MB https://www.dropbox.com/s/y60pwswbbch0lim/DBErrors.log?dl=0
       

### Steps to Reproduce

1. Start server 
2. Let server run
3. Server crashes after x period of days, depending on the amount of memory
4.

### Expected behavior

Memory not to leak

### Suggested Workaround

N/A

### Crash Log

[973870.309182] Out of memory: Killed process 110929 (mangosd) total-vm:66189104kB, anon-rss:47953568kB, file-rss:0kB, shmem-rss:0kB, UID:_XXXX_ pgtsables:126068kB oom_score_adj:0

./world-wotlk.sh: line 2: 110929 Killed        ./mangosd -c ../etc/mangosd.conf -a ../ahbot.conf ../etc/playerbot.conf

### Core SHA1 Commit Hash

Using commit hash(d96bb3a3bd0c5585603f352de2ca927190507465) committed on 2022-09-08T17:03:20+02:00

### Database SHA1 Commit Hash

2f69f12fb02f9b5466e5d14dac5457b9c344e451

### Operating System

Ubuntu 20.04.4 Server LTS

### Client Version

3.3.5a (Wrath of the Lich King)

**Comment by jrtashjian** [comment]
I have not investigated much yet and I am running each server variant using [container images I've put together](https://github.com/jrtashjian/cmangos-docker). I noticed the servers crashing and restarting (Docker is starting the failed container). Each graph below represents a single `realmd` and two `mangosd` instances running on a single virtual machine.

WotLK crashes around every 3 hours:

![image](https://user-images.githubusercontent.com/375788/193369796-1a8badc2-99a3-438a-9e3f-ad2582841395.png)

TBC is less consistent but still trends:

![image](https://user-images.githubusercontent.com/375788/193369853-0a018556-bd15-4a7e-9ce7-a6d69a3f0344.png)

Classic is much more stable but still trending up over time (no crashes yet):

![image](https://user-images.githubusercontent.com/375788/193369910-ee3abde6-4eb0-4a6a-9290-c376e774b9f7.png)

**Comment by Grz3s** [comment]
Hi @adellaci 
First let me know what log lvl you have in config file...
```code
LogLevel = ?
LogFileLevel = ?
```

Second
just looked at your ```DBErrors.log 578.8 MB https://www.dropbox.com/s/y60pwswbbch0lim/DBErrors.log?dl=0```
theyre tons of very strange DB errors there... let me show you few:
```code
Table `creature` has creature (GUID: 45221, Entry: 7856) with equipment_id 1261 already defined in creature_template table
```
WoTLKDB .. creature table for this guid (45221) has `0` in `equipment_id` .. so how your error shows smth diff???
```code
Table `creature` has creature (GUID: 5126, Entry: 771) with equipment_id 1189 already defined in creature_template table
```
same situation.... 

Lets take smth diff.
```code
Gameobject (Entry: 188067 GoType: 5) have too small size=0.000000
Gameobject (Entry: 188142 GoType: 5) have too small size=0.000000
Gameobject (Entry: 188073 GoType: 5) have too small size=0.000000
Gameobject (Entry: 188072 GoType: 5) have too small size=0.000000
Gameobject (Entry: 187882 GoType: 2) have too small size=0.000000
Gameobject (Entry: 188077 GoType: 1) have too small size=0.000000
Gameobject (Entry: 194752 GoType: 10) have too small size=0.000000
Gameobject (Entry: 202931 GoType: 5) have too small size=0.000000
```
all of them have diff size than 0 in WoTLkDB...

I had wotlk core running for whole day today... and never went above 1.5GB RAM
im not using BOTs and AH.. just to be fair.
![sfdsfłu](https://user-images.githubusercontent.com/2374069/193375735-a69431c3-ad75-46e0-a115-bb616b779d30.jpg)

I will continue checking your error log koz it contains some interesting stuff...

**Comment by adellaci** [comment]
@Grz3s here is my logging settings
```
LogLevel = 0
LogFileLevel = 1 
```
Core: 300d52d4b024bc7ac18627609c961b490aaea590
DB:    a7d47747d16e838454afd5934c2ee921c96e46d9

When install DB i selected option " 2) Install fresh WoTLK-DB only to 'wotlkmangos' "

Started server again as follows;
Just "realmd" running
Virt: 116652 KB
Res: 9120 KB
Shr: 8136 KB

Total System: 639.9 MB

After 'mangosd' started

```
./mangosd -c ../etc/mangosd.conf -a ../etc/ahbot.conf ../etc/playerbot.conf
```
With these parameters that have an impact for mem usage
```
GridUnload = 0
LoadAllGridsOnMaps = "DO"
```
mangosd

Virt  3067820 KB
Res 2.2 GB
Shr 26264 kB

Total System Usage:
MiB Mem :  48187.6 total,  42898.3 free,   3021.7 used,   2267.6 buff/cache


After 15 minute of running no one logged in;
MiB Mem :  48187.6 total,  41792.3 free,   3192.9 used,   3202.3 buff/cache
mangos    3198892kB            2.4GB        26276kb    mangosd                                                                                                                                     
mysql       3281964kB      520108kB        38628kb    mysqld                                                                                                                                      
mangos     116652kb           9120kB          8136kb    realmd

After 30 minute of running no one logged in;
MiB Mem :  48187.6 total,  41654.7 free,   3305.0 used,   3227.9 buff/cache
mangos  3395500kb             2.5GB          26276kB   mangosd                                                                                                                                     
mysql     3281964KB       520108KB         38628kB   mysqld                                                                                                                                      
mangos    116652KB          9120KK            8136kB   realmd

**Comment by adellaci** [comment]
After 330 minutes
MiB Mem :  48187.6 total,  39070.3 free,   5391.1 used,   3726.1 buff/cache
mangos    5558188KB        4.5GB         26276KB     mangosd                                                                                                                                     
mysql       3282996KB   520108KB         38628KB    mysqld                                                                                                                                      
mangos      116652KB      9120KB           8136KB    realmd

**Comment by adellaci** [comment]
After 19 hours

MiB Mem :  48187.6 total,  32030.1 free,  12083.9 used,   4073.5 buff/cache
mangos   13.6gB  11.1gB  26276kb mangosd

**Comment by adellaci** [comment]
@Grz3s I will update server and database, and go there the various combination with the two parameters;
GridUnload = 0
LoadAllGridsOnMaps = "DO"

**Comment by Grz3s** [comment]
why would you load everything even if no Players ae there...... its a waste of RAM... 
```
#    GridUnload
#        Unload grids (if you have lot memory you can disable it to speed up player move to new grids second time)
```
```
#        Load grids of maps at server startup (if you have lot memory you can try it to have a living world always loaded)
#        This also allow ALL creatures on the given maps to update their grid without any player around.
#        Default: "" (don't load all grids at startup)
#                 "mapId1[,mapId2[..]]" (DO load all grids on the given maps- Experimental and very resource consumming)
```
An you complaining about ram usage .... oh lol

**Comment by jrtashjian** [comment]
Just posting another update here. I checked my Grid configuration and it is set to the defaults.

```
GridUnload = 1
LoadAllGridsOnMaps = ""
```

I did notice from the description that maybe `Autoload.Active` might cause an increase in resource usages. The default value is `True` even though the description indicates the setting is "Useful for running in debug mode." So I tried disabling it:

```
Autoload.Active = 0
```

To my surprise the WotLK server has been stable and hasn't crashed (empty server) where it was crashing every ~3 hours previously (empty server). Here's the average memory usage graph over the past day and it's holding steady. Surprisingly CPU usages has also decreased from an average of 15% on 4-cores to 2% usage.

![Screenshot from 2022-10-09 19-44-05](https://user-images.githubusercontent.com/375788/194785064-735da448-1617-4a00-927d-95734cdcfc5d.png)

I'll get some debug logs posted with the `Autoload.Active` setting enabled and disabled soon.

**Comment by adellaci** [comment]
@Grz3s 
I have always ran it with those configuration. Before this started i had run it for months on end with out memory leak. Memory leaks have occurred in past but were fixed quickly. 
Typically my memory usage would not exceed 5G.

I will try @jrtashjian suggestion and see how it impacts it.

**Comment by adellaci** [comment]
And i thing i just realized before this started mangos would load in 23 sec or less. Now it takes a minute plus. They my be unrelated.

**Comment by Grz3s** [comment]
Yes but u need to give that info ... when u making a ticket.
Also i seriuosly see no reason to load stuff that happens for example on another continent... even if noone is even there...you only slowing whole thing down.. making your cpu/gpu consume more power etc for no reason. We understand that some of events need to be ACTIVE all the time so they are by default.(but im personaly trying to keep numbers as low as poss.)

**Comment by adellaci** [comment]
Core = ebccc6b4d900d774f08f17cbf099b0e82b52de1b
DB = 08c1e2ab89d8ca9ca73e5441bfb756731449a9a0
GridUnload = 1
LoadAllGridsOnMaps = ""
Autoload.Active = 0

Before starting mangosd:
```
System Total
CPU%  = 0.5%      Memory Usage = 1.5G

 CPU%          MEM%      VIRT       RES      TIME+        THR      Command
  0.0           1.3     3.19G      612M   03:56.92         48       mysqld
  0.2           0.0      114M     8.91M   00:06.76          5       realmd
```
After starting mangosd: 1 minute 2 sec load time
```
System Total
CPU%  = 1.1%      Memory Usage = 2.7G

 CPU%          MEM%      VIRT       RES      TIME+        THR      Command
  0.0           1.3     3.19G      614M   04:13.67         48       mysqld
  0.2           0.0      114M     8.91M   00:07.41          5       realmd
  5.2           2.4     1.77G     1.14G   00:58.12         15      mangosd
```

Post again tomorrow at 24 hours

**Comment by jrtashjian** [comment]
Here are the debug logs showing the difference between `Autoload.Active` being set to `True` and `False`. The setting is set to `True` by default and memory usage increases at a high rate where a `False` value is steady.

- [autoload-active-true_logs.zip](https://github.com/cmangos/issues/files/9759942/autoload-active-true_logs.zip)
- [autoload-active-false_logs.zip](https://github.com/cmangos/issues/files/9759943/autoload-active-false_logs.zip)

I'm going to submit a PR for each variant to change this default setting to `False`.

**Comment by Grz3s** [comment]
`Autoload.Active = 1` - loads creatures that have extraflag active `ONLY` .. and they have that flag for reason
by disabling it.. u stopping some important events like darkportal invasions etc...  Im not recommending that... 
They're less than 100 of them for whole database!

This ticket is about these 2:
```
GridUnload = 0 -- default is 1
LoadAllGridsOnMaps = "DO" -- default is ""
```
They were making `everything` loaded... so ~150K creatures + ~108K objects... see the difference?

**Comment by jrtashjian** [comment]
Yes, that makes sense to me. I do realize that the original issue is likely caused by loading everything with those settings defined as they are. I'm bringing the conversation back to the memory usage problem which is causing the crash regardless of the settings unveiled as the problem. In a default setup with the default settings I was seeing a consistent cycle in memory usage until the server crashes (every ~3 hours) when idle (no players logged in). The biggest positive change was changing `Autoload.Active` to `0` which, with the new information, I understand that's not the correct solution.

So my next step is to compile the server without BOTs and AH, with just BOTs, and with just AH. I'll monitor resource usage and see if we can figure out the culprit.

**Comment by Grz3s** [comment]
make sure these two are set to default:
```code
GridUnload = 1
LoadAllGridsOnMaps = ""
```

**Comment by adellaci** [comment]
@Grz3s  I did set it that way for this test, still running server.

After starting mangosd: 1 day 20 hours
```
System Total
CPU%  = 0.8%      Memory Usage = 3.0G

 CPU%          MEM%      VIRT       RES      TIME+        THR      Command
  0.0           1.3     3.19G      614M   10:55.23         49       mysqld
  0.2           0.0      114M     8.91M   02:06.44          5       realmd
  4.8           3.2     2.21G     1.50G    2h19:34         15      mangosd
```

**Comment by adellaci** [comment]
In this configuration the process is very stable. Memory is not being eaten up over time. I'll change the configuration and run it again.

After starting mangosd: 12 day 3 hours
```
System Total
CPU%  = 1.2%      Memory Usage = 3.7G

 CPU%          MEM%      VIRT       RES      TIME+        THR      Command
  0.0           1.3     3.19G      614M   10:55.23         49       mysqld
  0.2           0.0      114M     8.91M   02:06.44          5       realmd
  5.1          3.01     2.18G     1.50G   14h30:26         15      mangosd

**Comment by evil-at-wow** [comment]
> GridUnload = 0
> LoadAllGridsOnMaps = "DO"

I'm going to note that last line is most likely not doing what you intended (although after checking the comments in the default configuration file I can see where the confusion comes from...).

The core expects to find a comma-separated list of map IDs in ```LoadAllGridsOnMaps```, so "DO" is actually not a valid value. The core handles it as a comma-separated list though, with just one element in this case, and then tries to interpret this element (DO) as an integer. This kind of works, because the [API](https://en.cppreference.com/w/cpp/string/byte/atoi) it uses to do so always succeeds ("If no conversion can be performed, ​0​ is returned."), but as you can surely guess by now it returns 0 because "DO" is not actually representing an integer. 

So effectively, your value has the same effect as using ```LoadAllGridsOnMaps = "0"```, which only loads all grids for map 0 (Eastern Kingdoms), while your intention was most likely to have ```LoadAllGridsOnMaps = "0, 1, 530, 571"```. Yes, if you correct this in your config, the server will use even more memory to load everything! And, I guess, for now it will run out of memory even faster, because you have now 4 continents (3 in TBC) that may be continuously leaking some memory rather than just one...

**Comment by adellaci** [comment]
When I get back on Monday I will test it.

Dec 22, 2022 11:25:59 evil-at-wow ***@***.***>:

**Comment by adellaci** [comment]
Realm, Apache2, MySQL running
![Realm_Apache2_MySQL_Mem_2022-12-29 19-17-11](https://user-images.githubusercontent.com/13931954/210023112-9ff9911c-31ce-4eed-86db-e2e54d02d1f4.png)

**Comment by adellaci** [comment]
Realm, Apache2, MySQL, World after completely load
![Screenshot from 2022-12-29 19-27-41](https://user-images.githubusercontent.com/13931954/210023503-455f36ca-e419-4017-8b05-d8c13f47fcf1.png)

**Comment by adellaci** [comment]
> > GridUnload = 0
> > LoadAllGridsOnMaps = "DO"
> 
> I'm going to note that last line is most likely not doing what you intended (although after checking the comments in the default configuration file I can see where the confusion comes from...).
> 
> The core expects to find a comma-separated list of map IDs in `LoadAllGridsOnMaps`, so "DO" is actually not a valid value. The core handles it as a comma-separated list though, with just one element in this case, and then tries to interpret this element (DO) as an integer. This kind of works, because the [API](https://en.cppreference.com/w/cpp/string/byte/atoi) it uses to do so always succeeds ("If no conversion can be performed, ​0​ is returned."), but as you can surely guess by now it returns 0 because "DO" is not actually representing an integer.
> 
> So effectively, your value has the same effect as using `LoadAllGridsOnMaps = "0"`, which only loads all grids for map 0 (Eastern Kingdoms), while your intention was most likely to have `LoadAllGridsOnMaps = "0, 1, 530, 571"`. Yes, if you correct this in your config, the server will use even more memory to load everything! And, I guess, for now it will run out of memory even faster, because you have now 4 continents (3 in TBC) that may be continuously leaking some memory rather than just one...

You are correct, I loaded all and it doubled memory usage. I will leave it running, and see how fast it decays

**Comment by adellaci** [comment]
It crashed before it consumed the memory. It set it back to LoadAllGridsOnMaps = "" and see how long it will run

**Comment by adellaci** [comment]
Started
2022-12-30 01:21:07 Built on Dec 29 2022 at 23:02:12
2022-12-30 01:21:07 Built for Linux_x64 (little-endian)
2022-12-30 01:21:07 Using commit hash(7586d09663de4ce6826812d71dc9b86298ba9f43) committed on 2022-12-29 T10:45:28+01:00

Crashed at
2022-12-31 12:20:01 Next game event check in 480 seconds.

It was not because of "Out of Memory". Last night before bed i checked it and it was 2.68G/47.0G

**Comment by adellaci** [comment]
Start a new memory leak 
2023-04-27 23:58:11 Using commit hash(f7aec4af7c95914227d8fa7888379e5240e5c1fc) committed on 2023-04-15T00:39:20+02:00
It has been very stable going to change configuration to see if it leaks
```
GridUnload = 1
LoadAllGridsOnMaps = ""
Autoload.Active = 0
```
I will start with 'GridUnload = 0' & 'Autoload.Active = 1' first

**Comment by adellaci** [comment]
It ran 8 days before before shutting down (39.8Gb/48Gb). This time it was MySQL that stopped

Setting configuration back to;
```
GridUnload = 1
LoadAllGridsOnMaps = ""
Autoload.Active = 0
```

**Comment by evil-at-wow** [comment]
Thank you for reporting back every now and then! I would say 8 days is a start, especially considering it usually crashed after hours half a year ago, but we're clearly not yet where we want to be.

Just to make sure we have the correct idea of context here, the workload of that core over those 8 days was still running mostly idle, with occasionally one (or at most a few) players playing for a while, right?

On a related note, I finally managed to find time to get a Wrath core repository and merge a bunch of crash fixes and memory leak fixes in the past week (fixes that have been in the tbc core for a few months already), so there should be some new improvements again. But I'm certainly not claiming we're done now (I have added a few more to my list in the past few weeks).

**Comment by adellaci** [comment]
I started the server and played 4 or so hours. I check memory usage after. It only increased about 1Gb +, but nothing crazy. Had not played since.

I did a pulled this morning. In the process of compiling and update the DB, then I will start it back up after.

Review the system log, an update to mysql occured causing it to go down. mangosd kept running with SQL error "unable to connect"

**Comment by adellaci** [comment]
I started server up on 2023-06-04 13:23:41 EST with configuration setting 'GridUnload = 0' & 'AutoLoad.Active = 1'
WoTLK Core      219be0df551e7fde494819ca785bcfe306510e30
Database           558eecb24aca5a3a218bd05f62449a91aff8cc93
2023-06-04 13:23:41 Memory Usage = 2.17 GB
2023-06-05 22:20:01 Memory Usage = 7.69 GB
2023-06-07 22:40:02 Memory Usage = 14.6 GB
2023-06-08 09:38:01 Memory Usage = 16.1 GB
2023-06-10 03:08:01 Memory Usage = 22.5 GB
2023-06-11 09:40:01 Memory Usage = 26.1 GB

At this rate it will it will have consumed the memory in in 17 days and  about 4 hours

**Comment by adellaci** [comment]
I started server up on 2023-06-11 13:23:41 with configuration setting 'GridUnload = 0' & 'AutoLoad.Active = 1'
WoTLK Core 0ab920a701aa2b82952ef1c34dc722637f785709
Database 638e33c23d922c7acd32b3a90a8a245b5edd1a1d

Memory usage this time will only mangosd process, as realmd and MySQL seems to hold constant.
Server Startup Time 1 minute 4 Seconds

2023-06-11 12:10:05 Memory Usage = 1625 MB
2023-06-11 14:10:01 Memory Usage = 2116 MB
2023-06-14 22:58:01 Memory Usage = 13.1 GB
2023-06-16 01:30:01 Memory Usage = 16.2 GB
2023-06-16 13:01:01 Memory Usage = 17.9 GB
2023-06-19 22:50:01 Memory Usage = 29.1 GB

At this rate it will it will have consumed the memory in in 13 days and about 8 hours

Going to leave it running till mem failure to see if my math works out.

2023-06-22 22:38:01 Memory Usage = 41.1 GB
2023-06-23 20:50:01 Memory Usage = 43.4 GB
2023-06-25 11:40:01 Memory Usage = 45.5 GB

It end at about 14 days

**Comment by al3xc1985** [comment]
Any new tests @adellaci ? with latest core changes

**Comment by adellaci** [comment]
@al3xc1985, no but i will start one right now.

CMaNGOS-Core revision : 27999d818ce8b308e30a24526b7705335d5b3dbc
WoTLK-Database revision : b276c0f5b99b0daaca23edd4117eac5c1f73c960

Fresh Compile of Core
Fresh Installation of Database

cmake \~/cmangos/wotlk -DCMAKE_INSTALL_PREFIX=~/wow-servers/wotlk -DBUILD_EXTRACTORS=ON -DBUILD_AHBOT=ON -DBUILD_PLAYERBOT=ON -DPCH=0 -DDEBUG=0 -DBUILD_RECASTDEMOMOD=OFF -DBUILD_GIT_ID=0 ; make -j6

Base Memeory Usage with just Realmd running
Memory usage : 971 MB

GridUnload = 1
LoadAllGridsOnMaps = ""
Autoload.Active = 0

SERVER STARTUP TIME: 0 minutes 45 seconds
01/13/2024 11:01 EST
Memory usage : 2.23 GB
Realmd : 9096 KB
Mangosd : 1259 MB

**Comment by adellaci** [comment]
Error that showed up in Server Log;
024-01-13 15:58:52 ERROR:Table combat_condition entry 4523 has invalid worldstate expression 30517. Setting to 0.
2024-01-13 15:58:52 ERROR:Table combat_condition entry 5102 has invalid worldstate expression 34365. Setting to 0.
2024-01-13 15:58:52 ERROR:Table combat_condition entry 5375 has invalid worldstate expression 35631. Setting to 0.
2024-01-13 15:58:52 ERROR:Table combat_condition entry 5376 has invalid worldstate expression 35632. Setting to 0.

2024-01-13 15:59:29 ERROR:Table 'reference_loot_template' entry 12992 isn't reference id and not referenced from loot, and then useless. additional achievement criteria data (97 disabled).

2024-01-13 15:59:29 ERROR:Table 'reference_loot_template' entry 12991 isn't reference id and not referenced from loot, and then useless.
2024-01-13 15:59:29 ERROR:Table 'reference_loot_template' entry 12992 isn't reference id and not referenced from loot, and then useless.

The other ones are related to 'total loot chance > 100%'
in 'creature_loot_template' (117), 'gameobject_loot_template' (5), 'item_loot_template' (5), & 'prospecting_loot_template' (2)

**Comment by adellaci** [comment]
SERVER STARTUP TIME: 0 minutes 45 seconds
01/13/2024 11:01 EST Memory usage : 2.23 GB / Realmd : 9096 KB / Mangosd : 1259 MB 1 player 2 hours
01/14/2024 07:42 EST Memory usage : 2.49 GB / Realmd : 9096 KB / Mangosd : 1552 MB 1 player 2.5 hours
01/14/2024 10:12 EST Memory usage : 2.53 GB / Realmd : 9096 KB / Mangosd : 1588 MB

Crash after hunter trap triggered...Starting over

**Comment by al3xc1985** [comment]
THIS is great info. I mean it doesn't hop the memory usage wtih 2 - 6 Gb at once.. I mean a few hundreds of Mb is nothing

**Comment by adellaci** [comment]
@al3xc1985 yes it looks very promising. I will continue to run the server for a couple week to see, then try again with full load and see how behaves then

**Comment by adellaci** [comment]
2024-01-14 22:50:18 SERVER STARTUP TIME: 0 minutes 24 seconds
2024-01-14 22:50:18 EST Memory usage : 2.39 GB / Realmd : 9096 KB / Mangosd : 1394 MB Restart after crash
2024-01-15 16:14:38 EST Memory usage : 2.54 GB / Realmd : 9096 KB / Mangosd : 1576 MB
2024-01-16 17:15:20 EST Memory usage : 2.56 GB / Realmd : 9096 KB / Mangosd : 1600 MB
2024-01-16 16:02:38 EST Memory usage : 2.58 GB / Realmd : 9096 KB / Mangosd : 1614 MB
2024-01-17 16:39:33 EST Memory usage : 2.59 GB / Realmd : 9096 KB / Mangosd : 1622 MB
2024-01-18 16:52:48 EST Memory usage : 2.61 GB / Realmd : 9096 KB / Mangosd : 1628 MB
2024-01-21 07:10:22 EST Memory usage : 2.59 GB / Realmd : 9096 KB / Mangosd : 1635 MB

**Comment by adellaci** [comment]
@al3xc1985  What you think? I'm feel good about the mem leak. If i did my math right, i'm looking around 20 weeks + till i run out of 45,800 MB. Does that sound right? Would you like to run this longer?

**Comment by al3xc1985** [comment]
@adellaci u are awesome. Not even blizzzz doesn't run their server for that long without restart.
As far as I can say they restart their realms once a week.

I mean from huge ram usage to this point I am very very satisfied,

Taking in consederation how many crashes were fixed last year and mem leaks, we rock baby

**Comment by al3xc1985** [comment]
most likely more improvements will come in near future from this awesome development team that will boost the performance even more.

**Comment by evil-at-wow** [comment]
That's looking much much better than what it used to be, so I'm pleased to see this. It's clearly not entirely leak-free yet but in the last 3 days you're looking at like a loss of ~15MB a day. Even if you tripple that and round to say 50 MB/day it takes you a year of uptime to get to like 20 GB used which your server would handle no problem. I bet you want to rebuild/restart the server more often than that I guess :wink:

**Comment by adellaci** [comment]
@evil-at-wow @al3xc1985 You want me to close this issue?

**Comment by al3xc1985** [comment]
I let evil to decide

**Comment by evil-at-wow** [comment]
@adellaci I was actually thinking yesterday we can probably close this, yes. So if you also feel the original problem is adequately addressed, feel free to close it. We can always start a new issue if it gets worse again.

**Comment by adellaci** [comment]
Agree

**Comment by jrtashjian** [comment]
Just wanted to say thank you all! I've been following this and have seen the memory improvements while using my docker container.
